src/Redistribution.sol

```diff
@@ -400,119 +396,80 @@ contract Redistribution is AccessControl, Pausable {
             inProximity(entryProof1.proveSegment, currentRevealRoundAnchor, winner.depth),
             "witness is not in depth"
         );
-        inclusionFunction(entryProof1, x * 2);
+        inclusionFunction(entryProof1, indexInRC1 * 2);
         stampFunction(entryProof1);
         socFunction(entryProofLast);
 
         require(
             inProximity(entryProof2.proveSegment, currentRevealRoundAnchor, winner.depth),
             "witness is not in depth"
         );
-        inclusionFunction(entryProof2, y * 2);
+        inclusionFunction(entryProof2, indexInRC2 * 2);
         stampFunction(entryProof2);
         socFunction(entryProofLast);
 
-        checkOrder(x, y, entryProof1.proofSegments[0], entryProof2.proofSegments[0], entryProofLast.proofSegments[0]);
+        checkOrder(
+            indexInRC1,
+            indexInRC2,
+            entryProof1.proofSegments[0],
+            entryProof2.proofSegments[0],
+            entryProofLast.proofSegments[0]
+        );
 
         emit WinnerSelected(winner);
-
         PostageContract.withdraw(winner.owner);
     }
 
-    function winnerSelection() internal returns (Reveal memory winner_) {
+    function winnerSelection() internal returns (Reveal memory _winner) {
         require(currentPhaseClaim(), "not in claim phase");
-
         uint256 cr = currentRound();
-
         require(cr == currentRevealRound, "round received no reveals");
         require(cr > currentClaimRound, "round already received successful claim");
 
-        string memory truthSelectionAnchor = currentTruthSelectionAnchor();
-
-        currentSum = 0;
-        currentWinnerSelectionSum = 0;
-        bytes32 randomNumber;
-        uint256 randomNumberTrunc;
-
         bytes32 truthRevealedHash;
         uint8 truthRevealedDepth;
+        uint256 redundancy;
 
-        emit CountCommits(currentCommits.length);
-        emit CountReveals(currentReveals.length);
+        bytes32[revealersDepth] memory _frozenOverlays;
+        bytes32[revealersDepth] memory _slashedOverlays;
```

Comment:
> AFAIK the underscore prefix at variable names is for input arguments in Solidity by convention.

yes but not only for that, this are the cases where leading underscore is commonly used
**Private State Variables, Function Parameters, and Local Variables in Functions**
doesn't have to be but it's not uncommon. 
I do find it a bit inconsistent though as we didn't use that much so we could think about refactoring them as well as others in this contract to match style.

Reply:
> bigger problem is that this will allocate 32*32 bytes for each output where most of the indices won't be assigned to anything.

need to think about this, any suggestions?